main.py

```diff
@@ -0,0 +1,83 @@
+from fastapi import FastAPI, Depends, HTTPException
+from sqlalchemy.orm import Session
+
+import crud
+import schemas
+from database import SessionLocal
+
+app = FastAPI()
+
+def get_db() -> Session:
+    db = SessionLocal()
+    try:
+        yield db
+    finally:
+        db.close()
+
+
+@app.get("/")
+def root() -> dict:
+    return {"message": "Hello World"}
+
+
+@app.get("/authors/", response_model=list[schemas.AuthorList])
+def read_all_authors(
+        skip: int = 0,
+        limit: int = 10,
+        db: Session = Depends(get_db)
+) -> list[schemas.AuthorList]:
+    return crud.get_all_authors(db=db, skip=skip, limit=limit)
+
+@app.post("/authors/", response_model=schemas.AuthorDetail)
+def create_author(
+        author: schemas.AuthorCreate,
+        db: Session = Depends(get_db),
+) -> schemas.AuthorDetail:
+    db_author = crud.get_author_by_name(db=db, name=author.name)
+    if db_author:
+        raise HTTPException(
+            status_code=400,
+            detail="Author with such name already exists",
+        )
+
+    return crud.create_author(db=db, author=author)
+
+@app.get("/authors/{author_id}", response_model=schemas.AuthorDetail)
+def read_author(
+        author_id: int,
+        db: Session = Depends(get_db)
+) -> schemas.AuthorDetail:
+    author = crud.get_author(db=db, author_id=author_id)
+    if author is None:
+        raise HTTPException(status_code=404, detail="Author not found")
+    return author
+
+@app.get("/books/", response_model=list[schemas.Book])
+def read_all_books(
+        db: Session = Depends(get_db),
+        author_id: int | None = None,
+        skip: int = 0,
+        limit: int = 10,
+) -> list[schemas.Book]:
+    return crud.get_book_list(
+        db=db, author_id=author_id, skip=skip, limit=limit
+    )
+
+@app.get("/books/{book_id}/", response_model=schemas.Book)

```

Comment:
The paths for your routes are mostly consistent, but the trailing slash in "/books/{book_id}/" is a bit unusual compared to the other routes. It's more common to omit the trailing slash in RESTful APIs, so you might want to standardize this unless there's a specific reason for it.